File: java/user-management-policy/src/main/java/com/hexagonalchess/usermanagement/RemoveFriend.java
package com.hexagonalchess.usermanagement;

public class RemoveFriend {

    private final FriendshipRepository friendshipRepository;

    public RemoveFriend(FriendshipRepository friendshipRepository) {
        this.friendshipRepository = friendshipRepository;
    }

    public void execute(UserId ownerId, UserId friendId) {
        friendshipRepository.deleteByOwnerAndFriend(ownerId, friendId);
    }
}

File: python/user_management_policy/remove_friend.py
from user_management_policy.friendship_repository import FriendshipRepository
from user_management_policy.user_id import UserId


def make_remove_friend(friendship_repository: FriendshipRepository):
    def remove_friend(owner_id: UserId, friend_id: UserId) -> None:
        friendship_repository.delete_by_owner_and_friend(owner_id, friend_id)
    return remove_friend
